dvc/repo/diff.py

```diff
@@ -131,45 +130,47 @@ def _to_path(output):
             else os.path.join(str(output), "")
         )
 
-    def _to_checksum(output):
-        if on_working_fs:
-            return ostage(
-                repo.odb.local,
-                output.path_info,
-                repo.odb.local.fs,
-                "md5",
-                dry_run=True,
-            )[1].hash_info.value
-        return output.hash_info.value
-
     for output in repo.index.outs:
         if _exists(output):
             yield_output = targets is None or any(
                 output.path_info.isin_or_eq(target) for target in targets
             )
 
+            if on_working_fs:
+                _, obj = ostage(
+                    repo.odb.local,
+                    output.path_info,
+                    repo.odb.local.fs,
+                    "md5",
+                    dry_run=True,
+                    dvcignore=output.dvcignore,
+                )
```

Comment:
For the record: We really shouldn't be calling this in such a verbose form, but this and other points are related to https://github.com/iterative/dvc/issues/6594